cellprofiler/modules/measurecorrelation.py

```diff
@@ -235,20 +277,155 @@ def run_image_pair_images(self, workspace, first_image_name,
             #
             coeffs = lstsq(np.array((fi,np.ones_like(fi))).transpose(),si)[0]
             slope = coeffs[0]
+            result += [[first_image_name, second_image_name, "-", "Correlation","%.3f"%corr],
+                       [first_image_name, second_image_name, "-", "Slope","%.3f"%slope]]
+            # Orthogonal Regression for Costes' automated threshold
+            nonZero = (fi > 0) | (si > 0)
+            
+            xvar = np.var(fi[nonZero],axis=0,ddof=1)
+            yvar = np.var(si[nonZero],axis=0,ddof=1)
+            
+            xmean = np.mean(fi[nonZero], axis=0)
+            ymean = np.mean(si[nonZero], axis=0)
+            
+            z = fi[nonZero] + si[nonZero]
+            zvar = np.var(z,axis=0,ddof=1)
+            
+            covar = 0.5 * (zvar - (xvar+yvar))
+            
+            denom = 2 * covar
+            num = (yvar-xvar) + np.sqrt((yvar-xvar)*(yvar-xvar)+4*(covar*covar))
+            a = (num/denom)
+            b = (ymean - a*xmean)
+            
+            i = 1
+            while(i > 0.003921568627):
+                Thr_fi_c = i
+                Thr_si_c = (a*i)+b
+                combt = (fi < Thr_fi_c) | (si < Thr_si_c)
+                costReg = scistat.pearsonr(fi[combt], si[combt])
+                if(costReg[0] <= 0):
+                    break
+                i= i-0.003921568627
+                        
+            # Costes' thershold calculation
+            combined_thresh_c = (fi > Thr_fi_c) & (si > Thr_si_c)
+            fi_thresh_c = fi[combined_thresh_c]
+            si_thresh_c = si[combined_thresh_c]
+            tot_fi_thr_c = fi[(fi > Thr_fi_c)].sum()
+            tot_si_thr_c = si[(si > Thr_si_c)].sum()
+            
+            #Threshold as percentage of maximum intensity in each channel
+            thr_fi = self.thr.value * scistat.scoreatpercentile(fi,100.0) / 100
```

Comment:
A couple things about scoreatpercentile:

The docs for scoreatpercentile indicate that it will become obsolete in the future and that users should use np.percentile for Numpy 1.9+. So one approach would be to do something like:

    try:
        from numpy import percentile
    except ImportError:
        from scipy.stats import scoreatpercentile as percentile

Is scistat.scoreatpercentile(x, 100.0) the same as np.max(x)? If so, that should be the choice.